### PR TITLE
Update legal hold indicators when status changes

### PR DESCRIPTION
**Pull request by typfel**
## What's new in this PR?

- Hide/show legal hold indicators when legal hold status changes
- Support displaying legal hold details screen for a user

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/ConversationList/ConversationListTopBarViewController.swift [hunk 1]
```diff
@@ -138,14 +134,19 @@ final class ConversationListTopBarViewController: UIViewController {
         return accountView
     }
     
+    func updateLegalHoldIndictor() {
+        topBar?.rightView = ZMUser.selfUser().isUnderLegalHold ? createLegalHoldView() : nil
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/ConversationList/ConversationListTopBarViewController.swift]:
```suggestion
        topBar?.rightView = selfUser.isUnderLegalHold ? createLegalHoldView() : nil
```

**Comment by alexaubry** [comment]
@typfel Building should pass now that I fixed the base branch, if you try again.

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift [hunk 2]
```diff
@@ -190,7 +197,7 @@ extension GroupDetailsViewController {
     func presentLegalHoldDetails() {
         let viewController = LegalHoldDetailsViewController(conversation: conversation)
         
-        navigationController?.pushViewController(viewController, animated: true)
+        present(viewController.wrapInNavigationController(), animated: true)
```
Comment (alexaubry) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift]: If we present it from the group details view controller, is it still gonna be shown as a form sheet on iPad?

File: Wire-iOS/Sources/UserInterface/LegalHoldDetails/LegalHoldDetailsViewController.swift [hunk 3]
```diff
@@ -25,6 +25,22 @@ class LegalHoldDetailsViewController: UIViewController {
     fileprivate let collectionViewController: SectionCollectionViewController
     fileprivate let conversation: ZMConversation
     
+    convenience init?(user: ZMUser) {
+        if user.isSelfUser {
+            if let managedObjectContext = user.managedObjectContext {
+                self.init(conversation: ZMConversation.selfConversation(in: managedObjectContext)) // TODO jacob replace with UI friendly getter
```
Comment (alexaubry) [inline Wire-iOS/Sources/UserInterface/LegalHoldDetails/LegalHoldDetailsViewController.swift]: If this possible to do this at the moment, or should we leave this as TODO?

File: Wire-iOS/Sources/UserInterface/LegalHoldDetails/LegalHoldDetailsViewController.swift [hunk 4]
```diff
@@ -25,6 +25,22 @@ class LegalHoldDetailsViewController: UIViewController {
...
+                return
+            }
+        } else {
+            if let conversation = user.oneToOneConversation {
```
Comment (alexaubry) [inline Wire-iOS/Sources/UserInterface/LegalHoldDetails/LegalHoldDetailsViewController.swift]: Is there a case where we would use it for something else than the self user?

File: Wire-iOS/Sources/UserInterface/ConversationList/ConversationListTopBarViewController.swift [hunk 5]
```diff
@@ -138,14 +134,19 @@ final class ConversationListTopBarViewController: UIViewController {
...
+    
     func updateTitle() {
         guard let middleView = topBar?.middleView as? UILabel else { return }
         middleView.text = ZMUser.selfUser().name
     }
 
     @objc
     func presentLegalHoldInfo() {
-        ///TODO: present legalhold screen
+        guard let legalHoldDetailsViewControllers = LegalHoldDetailsViewController(user: ZMUser.selfUser()) else { return }
```
Comment (alexaubry) [inline Wire-iOS/Sources/UserInterface/ConversationList/ConversationListTopBarViewController.swift]: Typo `legalHoldDetailsViewControllers` → `legalHoldDetailsViewController`

File: Wire-iOS/Sources/UserInterface/LegalHoldDetails/LegalHoldDetailsViewController.swift (hunk 3 above)
Comment (typfel) [reply Wire-iOS/Sources/UserInterface/LegalHoldDetails/LegalHoldDetailsViewController.swift]: It's possible to do. Was just waiting for the other PRs to merge first.

File: Wire-iOS/Sources/UserInterface/LegalHoldDetails/LegalHoldDetailsViewController.swift (hunk 4 above)
Comment (typfel) [reply Wire-iOS/Sources/UserInterface/LegalHoldDetails/LegalHoldDetailsViewController.swift]: I can imagine using it from a user profile so I wanted to support other users as well.

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift (hunk 2 above)
Comment (typfel) [reply Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift]: There's no design for that but that seems reasonable to me.

File: Wire-iOS/Sources/UserInterface/LegalHoldDetails/LegalHoldDetailsViewController.swift (hunk 3 above)
Comment (typfel) [reply Wire-iOS/Sources/UserInterface/LegalHoldDetails/LegalHoldDetailsViewController.swift]: It needs change in wire-ios-sync-engine

**Pull request merged**